### PR TITLE
Quality control for Rain 

### PR DESCRIPTION
**Pull request by bethanclarke**
Added R-code for Outliers in Climatic Check Data Rain Dialog

### CODE REVIEW

**Comment by shadrackkibet** [comment]
Reviewing and testing this.

File: instat/static/InstatObject/R/Backend_Components/summary_functions.R [hunk 1]
```diff
@@ -501,6 +502,30 @@ summary_skewness_mc <- function(x, na.rm = FALSE, ...) {
...
+  }
+  if(bupperlimit){
+    Q3 + coef*exp(skewnessweight*MC)*IQR
+  } else {
+    Q1 - coef*exp(-skewnessweight*MC)*IQR
+  }
+}
+
+# kurtosis function
+summary_kurtosis <- function(x, na.rm = FALSE, type = 2, ...) {
```
Comment (shadrackkibet) [inline instat/static/InstatObject/R/Backend_Components/summary_functions.R]: I see this function is a duplicate

Comment (bethanclarke) [reply instat/static/InstatObject/R/Backend_Components/summary_functions.R]: I will delete this duplicate once you're finished testing

File: instat/static/InstatObject/R/Backend_Components/summary_functions.R [hunk 2]
```diff
@@ -501,6 +502,30 @@ summary_skewness_mc <- function(x, na.rm = FALSE, ...) {
   return(robustbase::mc(x, na.rm = na.rm))
 }
 
+# skewness outlier limit function
+summary_outlier_limit <- function(x, coef = 1.5, bupperlimit=TRUE, bskewedcalc=FALSE, skewnessweight = 4,na.rm = TRUE, ...){
```
Comment (shadrackkibet) [inline instat/static/InstatObject/R/Backend_Components/summary_functions.R]: Just minor formatting here;  `bupperlimit=TRUE, bskewedcalc=FALSE` to be  `bupperlimit = TRUE, bskewedcalc = FALSE` and a space  after every `,`

**Comment by shadrackkibet** [comment]
This is working great, just a few small issues to sort out on the designer.
1) The skewness weight label is not properly aligned with it's Numeric updown.
2) Tab order.

File: instat/dlgClimaticCheckDataRain.vb [hunk 3]
```diff
@@ -59,7 +59,9 @@ Public Class dlgClimaticCheckDataRain
         ucrChkDryMonth.SetText("Dry Month")
         ucrChkDryMonth.Enabled = False
         ucrChkOutlier.SetText("Outlier")
-        ucrChkOutlier.Enabled = False
+        ucrChkOutlier.Checked = False
```
[the remark this answers is not in the document]
Comment (bethanclarke) [reply instat/dlgClimaticCheckDataRain.vb]: Why do not need to specify where its checked or not?

**Comment by bethanclarke** [comment]
Thank you very much for testing this @shadrackkibet  

I will add the label, remove the duplicate skewness code and fix those issues with the designer soon

File: instat/dlgClimaticCheckDataRain.vb [hunk 4]
```diff
@@ -59,7 +59,9 @@ Public Class dlgClimaticCheckDataRain
         ucrChkDryMonth.SetText("Dry Month")
         ucrChkDryMonth.Enabled = False
         ucrChkOutlier.SetText("Outlier")
-        ucrChkOutlier.Enabled = False
+        ucrChkOutlier.Checked = False
+        ucrChkSkewedData.SetText("Skewed Data")
+        ucrChkSkewedData.Checked = False
```
[the remark this answers is not in the document]
Comment (shadrackkibet) [reply instat/dlgClimaticCheckDataRain.vb]: I see you already have `ucrChkSkewedData.SetRDefault("FALSE")` and setvalues when checked and unchecked so this checkbox will be unchecked by default.

**Comment by dannyparsons** [comment]
I've merged this so @rdstern can test before release, but suggested changes can still be made.

**Pull request merged**